### PR TITLE
Move coverage reports over to bazel

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

After the switch to bazel3 (#4071), we can finally move the coverage reporting over to bazel. The coverage report was the last go-related action which was done outside of bazel. This hase some very important advantages for the builder-image:

 * We don't depend on the `go` version inside the builder image anymore. Bazel is now the mandatory source for the go version in use, `go` inside the build container now only needs to be updated to allow compiling tools for `make generate`.
 * Nothing uses libvirt-devel anymore inside the build container for building. It is now only used for executing the unit tests, where "some" version of the libvirt-lib libraries needs to be present (removing the test dependency will follow in a separate PR).

Overall this means that we have to update the build container now mostly only to consume newer kubernetes-code-generators or to get another bazel version. This opens the door to update the build container now only when we want to vendor in k8s code from a new minor k8s release.

We can now:

 * establish a clean process for provisioning and publishing builder images in kubevirt/project-infra
 * build builder images for supported k8s version up-front and integrate them in kubevirt/kubevirt when ready
 * use the kubevirt/project-infra build-containers as base images in our CI jobs, dropping the builder-image nesting need (`make generate` still needs it, but it is the only one).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by rmohr** [comment]
/cc @dhiller 
/cc @jean-edouard 

@jean-edouard I think we discussed races in the test-suite in the past. They are addressed here finally.

File: hack/goveralls.sh [hunk 1]
```diff
@@ -1,6 +1,5 @@
 #!/usr/bin/env bash
+set -eo pipefail
 
-set -e
-
-./hack/coverage.sh
-goveralls -v -debug -service=${CI_NAME} -coverprofile=.coverprofile -ignore=$(find -regextype posix-egrep -regex ".*generated_mock.*\.go|.*swagger_generated\.go|.*openapi_generated\.go" -printf "%P\n" | paste -d, -s)
+bazel run //vendor/github.com/wadey/gocovmerge:gocovmerge -- $(cat | sed "s# # ${BUILD_WORKING_DIRECTORY}/#g" | sed "s#^#${BUILD_WORKING_DIRECTORY}/#") >coverprofile.dat
```
Comment (jean-edouard) [inline hack/goveralls.sh:4]: Wow, never seen a subshell consume stdin from its callers caller, but yeah that works :)
```
$ echo test | echo $( cat )
test
$
```

**Review by jean-edouard — APPROVED** [review]
/lgtm

**Review by dhiller — APPROVED** [review]
/lgtm

Still need more information on how to proceed on the general plan of moving images towards project-infra, maybe we can set up a dedicated meeting for this?

**Comment by rmohr** [comment]
> Still need more information on how to proceed on the general plan of moving images towards project-infra, maybe we can set up a dedicated meeting for this?

Sounds good. :+1:

**Comment by rmohr** [comment]
/hold

With race detection enabled the certificate tests run sometimes into timeouts.

**Comment by rmohr** [comment]
/hold cancel

Fixed the cert-test flakiness, I think now forever. Taking the reference time-point after the CA generation which was mostly the reason for the unpredicatability.

**Review by dhiller — APPROVED** [review]
/lgtm
/approve

**Pull request merged**